docs: clarify external documentation support and update example URL (#2704)
canonical/charmcraft: docs/howto/manage-charms.rst

.. meta::

>     :description: Learn about initializing, configuring, packing, and publishing Juju charms with Charmcraft and releasing channel revisions on Charmhub.

Reviewer: Please apply the [meta description guidelines](https://documentation.ubuntu.com/starflow/how-to/add-a-page-meta-description/#write-a-meta-description) to this. :)

Author: Updated, thank you!

.. _manage-charms:

.. meta::

>     :description: Learn how to manage the full life cycle of a Juju charm with Charmcraft. This tutorial covers initializing, configuring, packing, publishing to Charmhub, and managing channel revisions.

Reviewer: ```suggestion
    :description: How to manage the full life cycle of a Juju charm with Charmcraft. This guide covers initializing, configuring, packing, publishing to Charmhub, and managing channel revisions.
```

This is a how-to guide, not a tutorial. :)

Author: Yes, good catch. Thank you for pointing this out, it has been updated.

resources, actions, configurations, relations, and libraries is generated and
published automatically in respective tabs.

Charmhub supports both `Discourse <https://discourse.charmhub.io/>`__ topics

> and externally hosted documentation sites. With an externally-hosted site,

Reviewer: ```suggestion
and externally-hosted documentation sites. With an externally-hosted site,
```

resources, actions, configurations, relations, and libraries is generated and
published automatically in respective tabs.

To provide your main user documentation, include its URL in your charm's project file
under the :ref:`links.documentation <charmcraft-yaml-key-documentation>` key.

> Charmhub supports both `Discourse <https://discourse.charmhub.io/>`_ topics

Reviewer: ```suggestion
Charmhub supports both `Discourse <https://discourse.charmhub.io/>`__ topics
```

External links [should use the double underscore](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#external-links) so they don't become an unnamed reference.

Author: Updated, thank you!

and externally hosted documentation sites.

> .. note::

Reviewer: Please make this a regular paragraph, instead of a note.

Author: I have removed the note. I also reordered the "Add docs" section so that the instruction is right before the code snippet, and ends with "E.g.," like the other sections in the guide.

    When using an externally hosted site, Charmhub will display a **Read documentation** button
    that will redirect users to the specified URL. The **Description** tab will display your

>     charm's basic metadata summary.

Reviewer: ```suggestion
With an externally-hosted site, Charmhub displays a **Read documentation** button
that redirects users to the specified URL. The **Description** tab displays your
charm's basic metadata summary.
```

I advise against [using admonitions](https://documentation.ubuntu.com/starflow/how-to/starcraft-style-guide/#admonitions) and future tense when possible.

Author: Updated, thank you!
